### PR TITLE
webhook_test: use testify

### LINKED ISSUE #12865: Migrate tests from "api4/webhook_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `api4/webhook_test.go` file.

The expected way to migrate this is to go to the `api4/webhook_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19620

### PR DESCRIPTION
**Pull request by drekar**
#### Summary
Update `api4/webhook_test.go` to use `testify`.

#### Ticket Link
Fixes #12865 
[MM-19620](https://mattermost.atlassian.net/browse/MM-19620)

### CODE REVIEW

File: api4/webhook_test.go [hunk 1]
```diff
@@ -32,18 +32,9 @@ func TestCreateIncomingWebhook(t *testing.T) {
...
-	}
-
-	if rhook.UserId != th.SystemAdminUser.Id {
-		t.Fatal("user ids didn't match")
-	}
-
-	if rhook.TeamId != th.BasicTeam.Id {
-		t.Fatal("team ids didn't match")
-	}
+	require.Equal(t, rhook.ChannelId, hook.ChannelId, "channel ids didn't match")
```
Comment (cpoile) [inline api4/webhook_test.go]: Thanks for the submission @drekar ! 
For these tests, `require.Equal`'s signature is: `func Equal(t TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{})` -- so can you go through and make sure the `expected` and `actual` values are in the right position? 
Thanks!

Comment (drekar) [reply api4/webhook_test.go]: Sure thing, thanks for the review.

**Comment by drekar** [comment]
Rebasing off a master updated from upstream.

**Comment by deanwhillier** [comment]
Thanks for the contribution @drekar!

File: api4/webhook_test.go [hunk 2]
```diff
@@ -135,17 +126,11 @@ func TestGetIncomingWebhooks(t *testing.T) {
...
-	}
+	require.True(t, found, "missing hook")
 
 	hooks, resp = th.SystemAdminClient.GetIncomingWebhooks(0, 1, "")
 	CheckNoError(t, resp)
-
-	if len(hooks) != 1 {
-		t.Fatal("should only be 1")
-	}
+	require.Equal(t, len(hooks), 1, "should only be 1")
```
Comment (cpoile) [inline api4/webhook_test.go]: Awesome, thanks @drekar !
This should also be `expected, actual`. And a few more like this below. 
Other than that, looking good, almost ready.

**Comment by drekar** [comment]
Sorry @cpoile .  It looks like I got my wires crossed with @lucianomagrao. I had accepted issues #12860-12869, but for some reason the update didn't take on this one, but I did it anyway. I'm deleting my PR because @lucianomagrao's is also approved and predates mine.

**Pull request closed without merging**